aws_analysis_tools/cli/convert_ip.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+#
+# © 2016 Krux Digital, Inc.
+#
+
+#
+# Standard libraries
+#
+
+from __future__ import absolute_import
+from pprint import pformat
+
+#
+# Internal libraries
+#
+
+import krux_ec2.cli
+from krux_ec2.filter import Filter
+
+NAME = 'convert_ip'
+
+
+class Application(krux_ec2.cli.Application):
+
+    _IP = 'ip-address'
+    _PRIVATE_IP = 'private-ip-address'
+
+    def __init__(self, name=NAME):
+        # Call to the superclass to bootstrap.
+        super(Application, self).__init__(name=name)
+
+        # Set what AWS arg to filter based on
+        self.filter_arg = Application._PRIVATE_IP if self.args.private else Application._IP
+
+    def add_cli_arguments(self, parser):
+        # Call to the superclass first
+        super(Application, self).add_cli_arguments(parser)
+
+        parser.set_defaults(log_level='info')
+
+        group = krux_ec2.cli.get_group(parser, self.name)
+
+        group.add_argument(
+            "ip_address",
+            help="IP address to be converted",
+        )
+
+        group.add_argument(
+            "-p", "--private",
+            action='store_true',
+            default=False,
+            help="True if the given IP address is private. (default: %(default)s)",
+        )
+
+    def find_instance(self, filter_arg, address):
+        """
+        Searches for AWS instance based on given filter argument and ip address (private or normal).
+        """
+        f = Filter()
+        f.add_filter(
+            name=filter_arg,
+            value=address
+        )
+        return self.ec2.find_instances(f)
```

Comment:
I'll change the name, but I'd rather just keep it as a separate method to keep the run method cleaner.